### ISSUE TITLE
How to display database time in local time and customize the time format?

### ISSUE THREAD

**Issue by diwayou** [dbcodeio/public#85]
It would be great if the following features could be implemented:
1. The time in the database should be displayed as local time.
2. The time format should be customizable, for example, '2024-10-29 12:12:12'.

**Comment by mikeburgh** [comment]
I think both those might make sense as settings so people can adjust how they like.. 

I might need to also check on the formatting, as if you format it without a timezone, and then edit it, we loose that timezone data, and hence the DB may convert it based on it's timezone and end up being wrong..

**Comment by mikeburgh** [comment]
I just released version 1.4.14.<br><br>I worked through the dates types on all the currently supported DB's.

They will now display according to your computers locale setup, and when they contain timezone information it will be shown in your local time. <br><br>There is still a precision issue with fractional sections using the locale display, need to work on that, or add an option to allow ISO format. 

Let me know your thoughts once you try it out.

**Comment by zlyfer** [comment]
My issue is that the date column ('vertragsbeginn') is off by one day.

Additionally, the 'timestamp' column is displayed in multiple different time zones. I assume this happens because the timezone data is likely being retrieved from the values themselves or something similar. It’s probably my fault for having poor database practices, but I much prefer the way HeidiSQL presents the data.
I think having at least an optional setting for this would be a nice improvement.

This screenshot shows date and datetime values of two columns using the previous and updated version of dbcode and the values displayed by HeidiSQL.

![Image](https://github.com/user-attachments/assets/1dc5c684-8516-4f7f-86c5-e0a0af741ba3)

**Comment by mikeburgh** [comment]
I just pushed a new version 1.4.15

I don't think your DB practices are at fault, however you DB, the extension should handle it ;) 

Changes
- Setting to control display as ISO or Locale, defaults to ISO
- Setting to control if it shows in your computers local time or UTC (when timezone data is in the data), defaults to computers local time.
- Factional seconds will now be included in either style (ISO or Locale) 

With those two, you should be able to get it to match Heidi, with a couple of notes: 
- the timestamp column, the way heidi is showing those without any timezone.. I am not a fan of that as it's not clear they are actually UTC/Z. So in DBCode, it will put the Z on the end if you set it to show in UTC (or your timezone). That's what's happening with those GTM+1 and GMT+2 they are the same values as what Heidi is showing, just reflecting timezones. But with the above settings, if you set it to UTC, it should show the the way Heidi does (can you confirm that) ? 

- If present, fractional seconds will be displayed, in some cases that precision is important, don't want to loose it. 

- when the datatype does not have a timezone, we just show it as what the DB returns, since there is no timezone to base it off.

What I cannot reproduce though is that date being off between Heidi and DBCode.. I used DBViz, but could not replicate that at all.. will try Heidi tomorrow, but can you try the latest version and confirm it's still happening ?  It's clear it's just a date col, no time, or timezone based on the icon.. but I cannot replicate it being off by date in v1.4.14 and above.

**Comment by zlyfer** [comment]
Yep, can confirm it is still happening. Please let me know if there is any more information I can provide to help investigating this issue.

**Comment by mikeburgh** [comment]
Think I finally fixed this in 1.17.6, closing it now.
